### ISSUE TITLE
Getting error `ERROR:  subquery has too many columns`

### ISSUE THREAD

**Issue by michiomochi** [rails/rails#35418]
### Steps to reproduce

```
class AuditLog < ApplicationRecord
    belongs_to :actor, optional: true, class_name: 'User', foreign_key: 'user_id', inverse_of: :audit_logs
〜
```

```
class User < ApplicationRecord
    has_many :audit_logs, dependent: :nullify, inverse_of: :actor
〜
```

```
[15] pry(main)> AuditLog.where(actor: User.select(:id)).to_a
  AuditLog Load (2.6ms)  SELECT "audit_logs".* FROM "audit_logs" WHERE "audit_logs"."user_id" IN (SELECT "users"."id", "users"."id" FROM "users")
ActiveRecord::StatementInvalid: PG::SyntaxError: ERROR:  subquery has too many columns
LINE 1: ....* FROM "audit_logs" WHERE "audit_logs"."user_id" IN (SELECT...
                                                             ^
: SELECT "audit_logs".* FROM "audit_logs" WHERE "audit_logs"."user_id" IN (SELECT "users"."id", "users"."id" FROM "users")
from /Users/michiomochi/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activerecord-5.1.6.1/lib/active_record/connection_adapters/postgresql_adapter.rb:616:in `async_exec'
Caused by PG::SyntaxError: ERROR:  subquery has too many columns
LINE 1: ....* FROM "audit_logs" WHERE "audit_logs"."user_id" IN (SELECT...

[16] pry(main)> AuditLog.where(actor: User.select("id")).to_a
  AuditLog Load (10.9ms)  SELECT "audit_logs".* FROM "audit_logs" WHERE "audit_logs"."user_id" IN (SELECT "users"."id" FROM "users")
=> [#<AuditLog:0x00007ff9607c0ce8
  id: 1,
〜
```

`AuditLog.where(actor: User.select(:id))` is error.
But `AuditLog.where(actor: User.select("id"))` is no error.

### System configuration
**Rails version**: v5.1.6

**Ruby version**: 2.5.1

**Comment by kamipo** [comment]
This is fixed by #28713.

**Comment by michiomochi** [comment]
cool. thanks 🙏
